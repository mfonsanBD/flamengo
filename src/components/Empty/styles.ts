import styled, { css } from 'styled-components'

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
`

export const Image = styled.img`
  max-width: 100%;
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.red};
    font-size: ${theme.font.sizes.xxlarge};
    font-family: ${theme.font.family.googleSansMedium};
  `}
`

export const description = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.large};
    font-family: ${theme.font.family.googleSansMedium};
    margin-bottom: ${theme.spacings.medium};
  `}
`

export const Error = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.darkGray};
    font-size: 9rem;
    font-family: ${theme.font.family.montserrat};
  `}
`
