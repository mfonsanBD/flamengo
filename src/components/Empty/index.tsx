import Button from 'components/Button'
import Link from 'next/link'

import * as S from './styles'

export type EmptyProps = {
  title: string
  description: string
  hasLink?: boolean
}

const Empty = ({ title, description, hasLink }: EmptyProps) => (
  <S.Wrapper>
    <S.Error>404</S.Error>
    <S.Title>{title}</S.Title>
    <S.description>{description}</S.description>

    {hasLink && (
      <Link href="/" passHref>
        <Button as="a">Voltar a PÃ¡gina Inicial</Button>
      </Link>
    )}
  </S.Wrapper>
)

export default Empty
