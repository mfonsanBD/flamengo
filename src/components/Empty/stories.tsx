import { Story, Meta } from '@storybook/react'
import Empty, { EmptyProps } from '.'

export default {
  title: 'Empty',
  component: Empty,
  parameters: {
    backgrounds: {
      default: 'black'
    }
  },
  args: {
    title: 'Página Não Encontrada',
    description: 'A página que você está tentando acessa não existe.',
    hasLink: true
  }
} as Meta

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />
