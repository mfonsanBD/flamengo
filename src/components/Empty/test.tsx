import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/testes/helpers'

import Empty from '.'

const props = {
  title: 'A simple title',
  description: 'A simple description'
}

describe('<Empty />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Empty {...props} hasLink />)

    expect(
      screen.getByRole('heading', { name: /a simple title/i })
    ).toBeInTheDocument()

    expect(screen.getByText(/a simple description/i)).toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: /voltar a pÃ¡gina inicial/i })
    ).toHaveAttribute('href', '/')

    expect(container.parentElement).toMatchSnapshot()
  })

  it('should not render link when haslink not passed', () => {
    renderWithTheme(<Empty {...props} />)

    expect(
      screen.queryByRole('link', { name: /go back to store/i })
    ).not.toBeInTheDocument()
  })
})
