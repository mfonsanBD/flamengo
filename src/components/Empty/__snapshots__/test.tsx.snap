// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render correctly 1`] = `
.c4 {
  background: #F80032;
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: GoogleSans Medium,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.8rem;
  padding: 0.8rem 3rem;
}

.c4:hover {
  background: #c50028;
}

.c4:after {
  content: '';
  width: 100%;
  height: 3px;
  position: absolute;
  top: 3rem;
  display: block;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
}

.c2 {
  color: #F80032;
  font-size: 6.4rem;
  font-family: GoogleSans Medium,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

.c3 {
  color: #FAFAFA;
  font-size: 2.4rem;
  font-family: GoogleSans Medium,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  margin-bottom: 3rem;
}

.c1 {
  color: #545454;
  font-size: 9rem;
  font-family: Montserrat Bold,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
}

<body>
  <div>
    <main
      class="c0"
    >
      <h3
        class="c1"
      >
        404
      </h3>
      <h2
        class="c2"
      >
        A simple title
      </h2>
      <p
        class="c3"
      >
        A simple description
      </p>
      <a
        class="c4"
        href="/"
      >
        <span>
          Voltar a PÃ¡gina Inicial
        </span>
      </a>
    </main>
  </div>
</body>
`
